packages/salesforcedx-vscode-core/src/index.ts

```diff
@@ -699,3 +702,41 @@ export function deactivate(): Promise<void> {
   disposeTraceFlagExpiration();
   return turnOffLogging();
 }
+
+export async function validateCliInstallationAndVersion(): Promise<void> {
+  // Check that the CLI is installed and that it is a supported version
+  // If there is no CLI or it is an unsupported version then the Core extension will not activate
+  const c = new CheckCliVersion();
+
+  const sfdxCliVersionString = await c.getSfdxCliVersion();
+  const sfCliVersionString = await c.getSfCliVersion();
+
+  const sfdxCliVersionArray = await c.parseSfdxCliVersion(sfdxCliVersionString);
+  const sfCliVersionArray = await c.parseSfCliVersion(sfCliVersionString);
+
+  const cliInstallationResult = await c.validateCliInstallationAndVersion(sfdxCliVersionArray, sfCliVersionArray);
+
+  if (cliInstallationResult === CheckCliEnum.cliNotInstalled) {
+    showErrorNotification('sfdx_cli_not_found', [SFDX_CLI_DOWNLOAD_LINK, SFDX_CLI_DOWNLOAD_LINK]);
+    throw Error('No Salesforce CLI installed');
+  } else if (cliInstallationResult === CheckCliEnum.onlySFv1) {
+    showErrorNotification('sf_v1_not_supported', [SFDX_CLI_DOWNLOAD_LINK, SFDX_CLI_DOWNLOAD_LINK]);
+    throw Error('Only SF v1 installed');
+  } else if (cliInstallationResult === CheckCliEnum.outdatedSFDXVersion) {
+    showErrorNotification('sfdx_cli_not_supported', [SFDX_CLI_DOWNLOAD_LINK, SFDX_CLI_DOWNLOAD_LINK]);
+    throw Error('Outdated SFDX CLI version that is no longer supported');
+  } else if (cliInstallationResult === CheckCliEnum.bothSFDXAndSFInstalled) {
+    showErrorNotification('both_sfdx_and_sf', []);
+    throw Error('Both SFDX v7 and SF v2 are installed');
```

Comment:
are these user facing errors? if yes then can they also suggest the right course of action to take ?

Reply:
No, those are for telemetry purposes and are not user facing.  The user will see the error in the **showErrorNotification()** message, which you can find the text in **i18n.ts**.